app/src/mpris.c

```diff
@@ -0,0 +1,377 @@
+		g_dbus_method_invocation_return_value(invocation, NULL);
+	} else {
+        LOGW("mpris: unknown method %s", method_name);
+		g_dbus_method_invocation_return_error(invocation, G_DBUS_ERROR,
+											  G_DBUS_ERROR_UNKNOWN_METHOD,
+											  "Unknown method");
+	}
+}
+
+static GVariant *
+get_property_root(G_GNUC_UNUSED GDBusConnection *connection,
+				  G_GNUC_UNUSED const char *sender,
+				  G_GNUC_UNUSED const char *object_path,
+				  G_GNUC_UNUSED const char *interface_name,
+				  const char *property_name, G_GNUC_UNUSED GError **error,
+				  G_GNUC_UNUSED gpointer user_data) {
+	GVariant *ret;
+
+	printf("Getting property %s\n", property_name);
+	if (g_strcmp0(property_name, "CanQuit") == 0) {
+		ret = g_variant_new_boolean(FALSE);
+
+	} else if (g_strcmp0(property_name, "Fullscreen") == 0) {
+		int fullscreen = 0;
+		ret = g_variant_new_boolean(fullscreen);
+	} else if (g_strcmp0(property_name, "CanSetFullscreen") == 0) {
+		ret = g_variant_new_boolean(FALSE);
+	} else if (g_strcmp0(property_name, "CanRaise") == 0) {
+		ret = g_variant_new_boolean(TRUE);
+	} else if (g_strcmp0(property_name, "HasTrackList") == 0) {
+		ret = g_variant_new_boolean(FALSE);
+	} else if (g_strcmp0(property_name, "Identity") == 0) {
+		ret = g_variant_new_string("Identity");
+	} else if (g_strcmp0(property_name, "DesktopEntry") == 0) {
+		ret = g_variant_new_string("scrcpy");
+	} else if (g_strcmp0(property_name, "SupportedUriSchemes") == 0) {
+		ret = NULL;
+	} else if (g_strcmp0(property_name, "SupportedMimeTypes") == 0) {
+		ret = NULL;
+	} else {
+		ret = NULL;
+		g_set_error(error, G_DBUS_ERROR, G_DBUS_ERROR_UNKNOWN_PROPERTY,
+					"Unknown property %s", property_name);
+	}
+
+	return ret;
+}
+
+static gboolean
+set_property_root(G_GNUC_UNUSED GDBusConnection *connection,
+				  G_GNUC_UNUSED const char *sender,
+				  G_GNUC_UNUSED const char *object_path,
+				  G_GNUC_UNUSED const char *interface_name,
+				  const char *property_name, GVariant *value,
+				  G_GNUC_UNUSED GError **error, G_GNUC_UNUSED gpointer user_data) {
+	if (g_strcmp0(property_name, "Fullscreen") == 0) {
+		int fullscreen;
+		g_variant_get(value, "b", &fullscreen);
+        LOGD("mpris: setting fullscreen to %d", fullscreen);
+	} else {
+		g_set_error(error, G_DBUS_ERROR, G_DBUS_ERROR_UNKNOWN_PROPERTY,
+					"Cannot set property %s", property_name);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static GDBusInterfaceVTable vtable_root = {
+	method_call_root, get_property_root, set_property_root, {0}};
+
+static void
+method_call_player(G_GNUC_UNUSED GDBusConnection *connection,
+				   G_GNUC_UNUSED const char *sender,
+				   G_GNUC_UNUSED const char *_object_path,
+				   G_GNUC_UNUSED const char *interface_name,
+				   const char *method_name, G_GNUC_UNUSED GVariant *parameters,
+				   GDBusMethodInvocation *invocation, G_GNUC_UNUSED gpointer user_data) {
+	if (g_strcmp0(method_name, "Pause") == 0) {
+		g_dbus_method_invocation_return_value(invocation, NULL);
```

Comment:
Is there a reason the `Pause` method is not logged like the others?

```suggestion
	if (g_strcmp0(method_name, "Pause") == 0) {
        LOGD("mpris: Pause");
		g_dbus_method_invocation_return_value(invocation, NULL);
```